modules/ROOT/pages/6.4-release-notes.adoc

```diff
@@ -0,0 +1,1269 @@
+
+[9]: <https://github.com/tinymce/pillow/blob/main/CHANGELOG.md>
+
+[10]: <https://github.com/tinymce/pillow/commits/main>
+////
+
+
+For information on the **Enhanced Media Embed** plugin, see: xref:introduction-to-mediaembed.adoc[Introduction to Enhanced Media Embed].
+
+
+=== Permanent Pen 2.0.3
+
+The {productname} 6.4 release includes an accompanyin release of the **Permanent Pen** premium plugin.
+
+**Permanent Pen 2.0.3 includes the following fix:
+
+==== Updated the font and fontsize dropdown styles to match the default TinyMCE styles
+
+//#TINY-9630
+
+For information on the **Permanent Pen** plugin, see xref:permanentpen.adoc[Permanent Pen].
+
+
+=== Mentions 3.0.4
+
+The {productname} 6.4 release includes an accompanyin release of the **Mentions** premium plugin.
+
+**Mentions** 3.0.4 includes the following fix:
+
+==== It wasn’t possible to create a new line immediately after a pasted Mention.
+//#TINY-9432
+
+Previously, if a _Mention_ was added to a {productname} editor; and the _Mention_ was selected; copied or cut; and then pasted on to a new line, attempting to then create a new line by pressing the **Return** key failed.
+
+As well, and depending on the host browser, the insertion point would not present in the editor until it was moved, by clicking elsewhere in the editor.
+
+This could be worked-around by moving the insertion point and then moving it back to immediately after the _Mention_ or by typing any character and then deleting that character.
+
+With this update, pressing **Return** immediately following the pasting of a _Mention_ string in a {productname} editor creates a new line, as expected.
+
+Also, the insertion point displays in all browsers immediately following a pasted _Mention_ string, also as expected.
+
+For information on the **Mentions** plugin, see xref:mentions.adoc[Mentions].
+
+
+=== Spell Checker Pro 3.2.1
+
+The {productname} 6.4 release includes an accompanying release of the **Spell Checker Pro** premium plugin.
+
+**Spell Checker Pro** 3.2.1 includes the following fix.
+
+==== Spelling plugin can crash the service
+//#TINY-9391
+
+Previously, the Spell Checker Pro service could be crashed by large requests or by the Spell Checker Pro server not being able to manage the number of requests made in a given time. This latter issue could be exacerbated by the plugin immediately re-sending a failed query.
+
+With this update, Spell Checker Pro now caches outgoing and in-queue requests; filters duplicate requests; and retrieves returned responses for duplicate queries from the cache.
+
+This both limits and reduces the number of requests made to the Spell Checker Pro service, improving the reliability of the service and the plugin.
+
+For information on the **Spell Checker Pro** plugin, see: xref:introduction-to-tiny-spellchecker.adoc[Introduction to Spell Checker Pro].
```

Comment:
```suggestion
==== Spelling plugin can crash the service
//#TINY-9391

Previously, the Spell Checker Pro plugin would request each word in the content, and repeatedly request these words every time the spellchecker was activated. This resulted in duplicate words being requested during regular use of the plugin.

With this update, Spell Checker Pro now only sends requests to the spelling service for each unique word and caches these results for later retrieval. This greatly reduces the amount and size of requests sent by the plugin during regular use.

For information on the **Spell Checker Pro** plugin, see: xref:introduction-to-tiny-spellchecker.adoc[Introduction to Spell Checker Pro].
```
We don't want to describe how users can crash our service on old versions, this is a security vulnerability. Instead we should just outline the measures taken to reduce the volume and how it may affect self-hosted customers using the docker service (though that might not yet be documented).